### DISCUSSION TITLE
Delete babel runtime helpers in `@trpc/client`?

### DISCUSSION THREAD

**Discussion by KATT** [trpc/trpc#2232] [RFC / Ideas]
> Previously brought up by @mmkal

We could potentially remove babel runtime helpers in `@trpc/client`. The reason they are there is to add a transformation for `async`/`await` and other modern JS features.

However, it adds some unnecessary build times and complexity, maybe we can just get rid of it?

Is there even simple ways of people of transpiling the client themselves according to their browser requirements?

**Comment by mmkal** [comment]
I am by no means an expert on browser transpilation tools, but I think there are a lot of options these days. Starting with the simplest, `tsc` doesn't rely on `@babel/runtime` - when the target output doesn't include async/await, it generates some gross helper functions (which I think rely on `yield`) at the top of files that need it. I'm not sure about other popular transpilers like swc and esbuild, maybe they produce output which depends on `@babel/runtime`? But I'd hope not?

If it's useful, here's the result of running `microbundle` with no config on a demo project which uses async/await in its typescript (I missed the memo on moving off precronstruct which I think is similar - what was wrong with it?):

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/15040698/179399614-5ca9d6b4-3a8d-406e-85a6-6b6a99de51f6.png">

I think we'd just need to fiddle a bit to use the right `exports` configuration and then hopefully end-users' webpack configs would be able to do the right thing for various targets by default.

**Comment by mmkal** [comment]
Oh, and as a failsafe, we could add something to the docs for end-users who are targeting IE 11 and having trouble with it, to use [next-transpile-modules](https://www.npmjs.com/package/next-transpile-modules) to just transpile trpc's output code themselves. Not sure if it'd be necessary though.

**Comment by KATT** [comment]
Moved to #2314
